src/wast-parser.y

```diff
@@ -623,6 +708,29 @@ if_ :
     }
 ;
 
+rethrow_check : RETHROW {
+      if (!parser->flags->allow_exceptions) {
```

Comment:
I think I'd prefer having a function that does this check, rather than having it as part of the grammar. Does that break the error-handling?